gdx-pay-android-huawei/src/main/java/com/badlogic/gdx/pay/android/huawei/HuaweiPurchaseManager.java

```diff
@@ -224,29 +229,34 @@ public void onSuccess(PurchaseIntentResult result) {
                     // Obtain the payment result.
                     Status status = result.getStatus();
                     if (status.hasResolution()) {
-                        huaweiPurchaseManagerConfig.iapListener.onPurchaseResult(result);
+                        try {
+                            result.getStatus().startResolutionForResult(activity, PURCHASE_STATUS_RESULT_CODE);
+                        } catch (IntentSender.SendIntentException e) {
+                            huaweiPurchaseManagerConfig.observer.handlePurchaseError(e);
+                        }
                     }
                 }
             }).addOnFailureListener(new OnFailureListener() {
                 @Override
                 public void onFailure(Exception e) {
-                    if (e instanceof IapApiException) {
-                        IapApiException apiException = (IapApiException) e;
-                        huaweiPurchaseManagerConfig.iapListener.onIAPError(apiException);
-                    } else {
-                        huaweiPurchaseManagerConfig.iapListener.onError(e);
-                    }
-
                     huaweiPurchaseManagerConfig.observer.handlePurchaseError(e);
                 }
             });
         } else {
-            Exception e = new ProductInfoNotFoundException();
-            huaweiPurchaseManagerConfig.iapListener.onError(e);
-            huaweiPurchaseManagerConfig.observer.handlePurchaseError(e);
+            huaweiPurchaseManagerConfig.observer.handlePurchaseError(new ProductInfoNotFoundException());
```

Comment:
better use FetchItemInformationException, this is what other implementations give to the client